File: Python/classes.py
class Automobile:
  def __init__(self, year, make, model):
    self.year = year
    self.make = make
    self.model = model

  def revIt(self):
    print("vroom vroom")

class Truck(Automobile):
  def __init__(self, year, make, model):
    super().__init__(year, make, model)
    self.payload_tons = 10

auto1 = Automobile(2021, "Ford", "Fiesta")

print(auto1.year)
print(auto1.make)
print(auto1.model)

redTruck = Truck(2022, "Tesla", "Super Truck")

redTruck.revIt()
print(redTruck.payload_tons)

File: Javascript/classes.js



class Automobile {
    constructor(year, make, model) {
      this.year = year;
      this.make = make;
      this.model = model;
    }

    
}

// unnamed class
class Truck extends Automobile {
    constructor(year, make, model) {
        super(year, make, model);
      }

     revIt() {
        console.log(`${this.model} goes VROOOOOOMMMM.`); 
     } 
  };


  let SportsCar = class FastAutomobile {
    constructor(year, make, model) {
        this.year = year;
        this.make = make;
        this.model = model;
      }

    revIt() {
        console.log(`${this.model} goes vroom vroom.`);
    }
  };


  let porsche = new SportsCar(1992, "porsche", "911")
  console.log(porsche.model);
  porsche.revIt();

  

  function Person (name) {
    this.name = name;
  }
  
  Person.prototype.speak = function () {
    console.log(`${this.name} says Hello.`);
  }